Assets/Scripts/CoinManagerScript.cs

```diff
@@ -0,0 +1,105 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using DG.Tweening;
+
+public class CoinManagerScript : MonoBehaviour
+{
+
+    [SerializeField] GameObject coinPrefab;
+    [SerializeField] Transform coinTarget;
+    [SerializeField] Text textField;
+
+    [Space]
+    [Header("Available coins: (coins to pool)")]
+    [SerializeField] int maxCoins;
+
+    [Space]
+    [Header("Animation settings:")]
+    [SerializeField] [Range(0.5f, 2.9f)] float minAnimationDuration;
+    [SerializeField] [Range(2.9f, 0.5f)] float maxAnimationDuration;
+    [SerializeField] Ease easeType;
+    [SerializeField] float spread;
+
+    private Queue<GameObject> coinsQueue = new Queue<GameObject>();
+    private Vector3 targetPosition;
+    private int _c = 0;
+
+    public int Coins
+    {
+        get { return _c; }
+        set
+        {
+            _c = value;
+
+            // set the text value
+            textField.text = $"x{_c}";
+        }
+    }
+
+    void Awake()
+    {
+        targetPosition = coinTarget.position;
+
+        //prepare pool
+        PrepareCoins();
+
+    }
+
+    void PrepareCoins()
+    {
+        GameObject coin;
+        for (int i = 0; i < maxCoins; i++)
+        {
+            coin = Instantiate(coinPrefab);
```

Comment:
Just curious, why are you defining GameObject coin before the for loop? Does that save memory or something?

Reply:
Hmm, yea Im not actually sure.  I was watching a tutorial and this was the style they went with.  I cant think of a way that this would save on memory.